snippets/go/src/nb_jobs/get-nb-jobs.go

```diff
@@ -0,0 +1,230 @@
+//This script can be run using NetBackup 8.2 and higher.
+//It gets all NetBackup services available on the given host.
+
+package main
+
+import (
+    "bytes"
+    "flag"
+    "fmt"
+    "log"
+    "os"
+    "crypto/tls"
+    "encoding/json"
+    "io/ioutil"
+    "net/http"
+    "net/http/httputil"
+)
+
+const(
+
+    port              = "1556"
+    policiesUri       = "config/policies/"
+    contentTypeV2       = "application/vnd.netbackup+json;version=2.0"
+    contentTypeV3     = "application/vnd.netbackup+json;version=3.0"
+    testPolicyName    = "vmware_test_policy"
+    testClientName    = "MEDIA_SERVER"
+    testScheduleName  = "vmware_test_schedule"
+)
+
+
+//###################
+// Global Variables
+//###################
+var (
+    nbmaster            = flag.String("nbmaster", "", "NetBackup Master Server")
+    username            = flag.String("username", "", "User name to log into the NetBackup webservices")
+    password            = flag.String("password", "", "Password for the given user")
+    domainType          = flag.String("domainType", "", "Domain type of the given user")
+)
+
+const usage = "\n\nUsage: go run ./get-nb-jobs.go -nbmaster <masterServer> -userName <username> -password <password>\n\n"
+
+func main() {
+    // Print usage
+    flag.Usage = func() {
+        fmt.Fprintf(os.Stderr, usage)
+        os.Exit(1)
+    }
+
+    // Read command line arguments
+    flag.Parse()
+
+    if len(*nbmaster) == 0 {
+        log.Fatalf("Please specify the name of the NetBackup Master Server using the -nbmaster parameter.\n")
+    }
+    if len(*username) == 0 {
+        log.Fatalf("Please specify the username using the -username parameter.\n")
+    }
+    if len(*password) == 0 {
+        log.Fatalf("Please specify the password using the -password parameter.\n")
+    }
+
+    httpClient := GetHTTPClient()
+
+    jwt := Login(*nbmaster, httpClient, *username, *password)
+
+    //gives all the netbackup jobs
+    fmt.Printf("====================================================\n")
+    fmt.Printf("Gives list of all Netbackup jobs\n")
+    GetNetbackupJobs(httpClient, jwt, *nbmaster)
+    fmt.Printf("====================================================\n")
+
+   //to get all the netbackup jobs whose jobType is 'Backup'
+    fmt.Printf("====================================================\n")
+    fmt.Printf("Gives list of all Netbackup jobs of jobType 'Backup'\n")
+    GetBackupJobs(httpClient, jwt, *nbmaster)
+    fmt.Printf("====================================================\n")
+}
+
+
+//##############################################################
+// Setup the HTTP client to make NetBackup Policy API requests
+//##############################################################
+func GetHTTPClient() *http.Client {
+    tlsConfig := &tls.Config {
+        InsecureSkipVerify: true, //for this test, ignore ssl certificate
```

Comment:
Not just for this test. It's pretty standard to disable certificate validation in _all_ NetBackup communication because the certificate is self-signed and thus unverifiable anyway. I think you can just remove the comment because it's not really indicating anything special about _this_ code.